src/towncrier/test/test_build.py

```diff
@@ -82,6 +83,35 @@ def test_command(self):
     def test_subcommand(self):
         self._test_command(_main)
 
+    def test_no_newsfragment_directory(self):
+        """
+        A missing newsfragment directory acts as if there are no changes.
+        """
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            setup_simple_project(create_newsfragments_directory=False)
+
+            result = runner.invoke(_main, ["--draft", "--date", "01-01-2001"])
+
+        self.assertEqual(1, result.exit_code, result.output)
+        self.assertIn("Failed to list the news fragment files.\n", result.output)
+
+    def test_no_newsfragments(self):
+        """
+        An empty newsfragment directory acts as if there are no changes.
+        """
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            setup_simple_project(create_newsfragments_directory=True)
+
+            result = runner.invoke(_main, ["--draft", "--date", "01-01-2001"])
+
+        # This should fail
+        self.assertEqual(result.exit_code, 0)
```

Comment:
I am not sure what is going on here.

I guess that this is just a TODO due to some strange code in Click, and in the change we will have a non-zero exit code here.

```suggestion
        # For now, exit code is zero, but it should be non-zero.
        # See: LINK TO ISSUE
        self.assertEqual(result.exit_code, 0)
```

Reply:
Let's disregard my comment, I'll remove it.  Do you want the pre-existing behavior of creating output with 'no significant changes' as the text and a zero exit code?  Or, are we just confusing each other starting with my bad comment I left laying around.

Reply:
Looks like the comment may have been a copy/paste error from the test after this.